source/d3d11/draw_call_tracker.cpp

```diff
@@ -1,105 +1,168 @@
 #include "draw_call_tracker.hpp"
 #include "runtime.hpp"
 #include <math.h>
+#include "log.hpp"
 
 namespace reshade::d3d11
 {
 	void draw_call_tracker::merge(const draw_call_tracker& source)
 	{
-		_counters.vertices += source.vertices();
-		_counters.drawcalls += source.drawcalls();
+		_global_counter.vertices += source.total_vertices();
+		_global_counter.drawcalls += source.total_drawcalls();
 
-		for (auto source_entry : source._counters_per_used_depthstencil)
+		for (auto &[depthstencil, snapshot] : source._counters_per_used_depthstencil)
 		{
-			const auto destination_entry = _counters_per_used_depthstencil.find(source_entry.first);
+			const auto destination_entry = _counters_per_used_depthstencil.find(depthstencil);
 
-			if (destination_entry == _counters_per_used_depthstencil.end())
-			{
-				_counters_per_used_depthstencil.emplace(source_entry.first, source_entry.second);
-			}
-			else
-			{
-				destination_entry->second.vertices += source_entry.second.vertices;
-				destination_entry->second.drawcalls += source_entry.second.drawcalls;
-			}
+			_counters_per_used_depthstencil[depthstencil].counter.vertices += snapshot.counter.vertices;
+			_counters_per_used_depthstencil[depthstencil].counter.drawcalls += snapshot.counter.drawcalls;
+		}
+
+		for (auto &[buffer, snapshot] : source._counters_per_constant_buffer)
+		{
+			const auto destination_entry = _counters_per_constant_buffer.find(buffer);
+
+			_counters_per_constant_buffer[buffer].vertices += snapshot.vertices;
+			_counters_per_constant_buffer[buffer].drawcalls += snapshot.drawcalls;
+			_counters_per_constant_buffer[buffer].pixelshaders += snapshot.pixelshaders;
+			_counters_per_constant_buffer[buffer].vertexshaders += snapshot.vertexshaders;
 		}
 	}
 
 	void draw_call_tracker::reset()
 	{
-		_counters.vertices = 0;
-		_counters.drawcalls = 0;
+		_global_counter.vertices = 0;
+		_global_counter.drawcalls = 0;
 		_counters_per_used_depthstencil.clear();
+		_counters_per_constant_buffer.clear();
 	}
 
-	void draw_call_tracker::log_drawcalls(UINT drawcalls, UINT vertices)
-	{
-		_counters.vertices += vertices;
-		_counters.drawcalls += drawcalls;
-	}
-	void draw_call_tracker::log_drawcalls(ID3D11DepthStencilView* depthstencil, UINT drawcalls, UINT vertices)
-	{
-		assert(depthstencil != nullptr && drawcalls > 0);
+	void draw_call_tracker::on_draw(ID3D11DeviceContext *context, UINT vertices) {
+		_global_counter.vertices += vertices;
+		_global_counter.drawcalls += 1;
 
-		const auto counters = _counters_per_used_depthstencil.find(depthstencil);
+		ID3D11RenderTargetView *rendertargets[D3D11_SIMULTANEOUS_RENDER_TARGET_COUNT] = { nullptr };
+		ID3D11DepthStencilView *depthstencil = nullptr;
 
-		if (counters != _counters_per_used_depthstencil.end())
+		context->OMGetRenderTargetsAndUnorderedAccessViews(D3D11_SIMULTANEOUS_RENDER_TARGET_COUNT, rendertargets, &depthstencil, 0, 0, nullptr);
```

Comment:
Seems like `OMGetRenderTargets` would do the job here as well, since the UAVs aren't used.